addons/purchase/models/account_invoice.py

```diff
@@ -214,7 +214,8 @@ def _anglo_saxon_purchase_move_lines(self, i_line, res):
     def create(self, vals):
         invoice = super(AccountInvoice, self).create(vals)
         purchase = invoice.invoice_line_ids.mapped('purchase_line_id.order_id')
-        if purchase and not invoice.refund_invoice_id:
+        recurrent_invoice = self.env.context.get('recurrent_invoice')
```

Comment:
Can't you rather use 'is_recurring_document' key, in vals ? This would remove the need for the context key.